python/lsst/pipe/tasks/make_psf_matched_warp.py

```diff
@@ -0,0 +1,215 @@
+            `psf_matched_warp`.
+        """
+        modelPsf = self.config.modelPsf.apply()
+
+        # Prepare the output exposure. We clone the input image to keep the
+        # metadata, but reset the image and variance plans.
+        exposure_psf_matched = direct_warp.clone()
+        exposure_psf_matched.image.array[:, :] = np.nan
+        exposure_psf_matched.variance.array[:, :] = np.inf
+        exposure_psf_matched.setPsf(modelPsf)
+
+        bit_mask = direct_warp.mask.getPlaneBitMask("NO_DATA")
+        total_good_pixels = 0  # Total number of pixels copied to output.
+
+        for row in direct_warp.info.getCoaddInputs().ccds:
+            transform = makeWcsPairTransform(row.wcs, direct_warp.wcs)
+            warp_psf = WarpedPsf(row.getPsf(), transform)
+
+            if (destination_polygon := row.validPolygon) is None:
+                # Calculate the polygon for this detector.
+                src_polygon = Polygon(
+                    [geom.Point2D(corner) for corner in row.getBBox().getCorners()]
+                )
+                destination_polygon = src_polygon.transform(transform).intersectionSingle(
+                    geom.Box2D(direct_warp.getBBox())
+                )
+                self.log.trace("Polygon for detector=%d is calculated as %s",
+                               row["ccd"],
+                               destination_polygon
+                               )
+            else:
+                self.log.trace("Polygon for detector=%d is read from the input calexp as %s",
+                               row["ccd"],
+                               destination_polygon
+                               )
+
+            # Compute the minimum possible bounding box that overlaps the CCD.
+            # First find the intersection polygon between the per-detector warp
+            # and the warp bounding box.
+            bbox = geom.Box2I()
+            for corner in destination_polygon.getVertices():
+                bbox.include(geom.Point2I(corner))
+            bbox.clip(direct_warp.getBBox())  # Additional safeguard
+
+            self.log.trace("PSF-matching CCD %d with bbox %s", row["ccd"], bbox)
+
+            ccd_mask_array = ~(destination_polygon.createImage(bbox).array <= 0)
+
+            # Clone the subimage, set the PSF to the model and reset the planes
+            # outside the detector.
+            temp_warp = direct_warp[bbox].clone()
+            temp_warp.setPsf(warp_psf)
+            temp_warp.image.array *= ccd_mask_array
+            temp_warp.mask.array |= (~ccd_mask_array) * bit_mask
+            # We intend to divide by zero outside the detector to set the
+            # per-pixel variance values to infinity. Suppress the warning.
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", message="divide by zero", category=RuntimeWarning)
+                temp_warp.variance.array /= ccd_mask_array
```

Comment:
Wanted to double check if this is an anti-pattern or fine as is.

Reply:
If you're sure it makes `inf` rather than `nan` it doesn't bother me too much; I'm annoyed by the fact that numpy decided this should be a warning.  But it is worth checking that the results are indeed `inf`.